cortex/connection.py

```diff
@@ -25,47 +25,55 @@
+
+    URIs = {"connections": "projects/{projectId}/connections"}
+
+    def save_connection(self, connection: dict) -> dict:
+        """Saves the provided `connection` dictionary to the Fabric API server.
+
+        :param connection: A Python dictionary containing specification for one of the `supported Fabric connection types <https://cognitivescale.github.io/cortex-fabric/docs/reference-guides/connection-types>`_
+        :type connection: dict
+        :return: A dictionary containing metadata about the saved connection
+        :rtype: dict
+        """  # pylint: disable=line-too-long
+        uri = self.URIs["connections"].format(projectId=self._project())
         data = json.dumps(connection)
-        headers = {'Content-Type': 'application/json'}
-        r = self._serviceconnector.request('POST', uri, data, headers)
-        raise_for_status_with_detail(r)
-        return r.json()
-
-    def get_connection(self, name: str):
-        """
-        Fetches a Connection to work with.
-        :param name: The name of the connection to retrieve.
-        :return: A Connection object.
-        """
-        port = os.getenv('CORTEX_CONNECTIONS_SERVICE_PORT_HTTP_CORTEX_CONNECTIONS') or '4450'
-        conn_svc_url = f'{self._serviceconnector.url.replace("cortex-internal", "cortex-connections")}:{port}'
-        uri = f'{conn_svc_url}/internal/projects/{self._project()}/connections/{urllib.parse.quote(name, safe="")}'
-        log.debug('Getting connection using URI: %s' % uri)
-        r = self._serviceconnector.request('GET', uri=uri, is_internal_url=True)
-        raise_for_status_with_detail(r)
-
-        return r.json()
+        headers = {"Content-Type": "application/json"}
+        res = self._serviceconnector.request("POST", uri, data, headers)
+        raise_for_status_with_detail(res)
+        return res.json()
+
+    def get_connection(self, name: str) -> dict:
+        """Fetch the specific connection in the `name` argument and return the details as a python dictionary
+
+        :param name: Name of the connection to be fetched from the Fabric API
+        :type name: str
+        :return: A python dictionary containing the specified connection details
+        :rtype: dict
+        """  # pylint: disable=line-too-long
+        port = (
+            os.getenv("CORTEX_CONNECTIONS_SERVICE_PORT_HTTP_CORTEX_CONNECTIONS")
+            or "4450"
+        )
+        conn_svc_url = f'{self._serviceconnector.url.replace("cortex-internal", "cortex-connections")}:{port}'  # pylint: disable=line-too-long
+        uri = f'{conn_svc_url}/internal/projects/{self._project()}/connections/{urllib.parse.quote(name, safe="")}'  # pylint: disable=line-too-long
+        log.debug("Getting connection using URI: {}", uri)
+        res = self._serviceconnector.request("GET", uri=uri, is_internal_url=True)
+        raise_for_status_with_detail(res)
+
+        return res.json()
 
     def _bootstrap(self):
```

Comment:
We might be able to kill this ? not sure this is a needed thing anymore ..

Reply:
We have a bunch fo public examples that still rely on this method. LEaving it in for now

Reply:
There are a few skills that still use this method.Do not recommend we drop this right now. Although since this is accessible from within a cluster only, it does make this unwieldy